rai-sdk/src/main/java/com/relationalai/TransactionAsyncResult.java

```diff
@@ -4,15 +4,32 @@
 
 public class TransactionAsyncResult extends Entity {
 
+    public Boolean gotCompletedResult;
     public TransactionAsyncCompactResponse transaction;
     public List<ArrowRelation> results;
     public List<TransactionAsyncMetadataResponse> metadata;
     public List<Object> problems;
 
-    public TransactionAsyncResult(TransactionAsyncCompactResponse transaction, List<ArrowRelation> results, List<TransactionAsyncMetadataResponse> metadata, List<Object> problems) {
+    public TransactionAsyncResult(
+            TransactionAsyncCompactResponse transaction,
+            List<ArrowRelation> results,
+            List<TransactionAsyncMetadataResponse> metadata,
+            List<Object> problems
+    ) {
         this.transaction = transaction;
         this.results = results;
         this.metadata = metadata;
         this.problems = problems;
+        this.gotCompletedResult = false;
+    }
+    public TransactionAsyncResult(
+            TransactionAsyncCompactResponse transaction,
+            List<ArrowRelation> results,
+            List<TransactionAsyncMetadataResponse> metadata,
+            List<Object> problems,
+            Boolean gotCompletedResult
+    ) {
+        this(transaction, results, metadata, problems);
+        this.gotCompletedResult = false;
```

Comment:
```suggestion
        this(transaction, results, metadata, problems, gotCompletedResult);
```
Otherwise we're not respecting what the user passed in as `gotCompletedResult`